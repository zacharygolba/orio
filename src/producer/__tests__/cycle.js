// @flow

import CycleProducer from '../cycle'

const SOURCE = 'test'
let producer

beforeEach(() => {
  producer = new CycleProducer(SOURCE)
})

test('#constructor()', () => {
  expect(producer).toMatchSnapshot()
})

test('#@@iterator()', () => {
  let i = 0

  for (const item of producer) {
    expect(SOURCE).toContain(item)

    if (i++ > SOURCE.length * 3) {
      break
    }
  }
})

test('#sizeHint()', () => {
  expect(producer.sizeHint()).toBe(Infinity)
})
