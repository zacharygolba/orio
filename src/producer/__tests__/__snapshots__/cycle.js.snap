// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#constructor() 1`] = `
CycleProducer {
  "source": "test",
  "state": 0,
}
`;
